Pod/SQL/GDGEntity+SQL.m

```diff
@@ -0,0 +1,246 @@
+//
+//  GDGEntity+SQL.m
+//  GoldDigger
+//
+//  Created by Felipe Lobo on 5/4/16.
+//
+
+#import <ObjectiveSugar/ObjectiveSugar.h>
+#import <objc/runtime.h>
+#import "GDGEntity+SQL.h"
+#import "SQLEntityQuery.h"
+#import "SQLEntityMap.h"
+#import "SQLTableSource.h"
+#import "GDGColumn.h"
+#import "GDGCondition.h"
+#import "GDGRelation.h"
+#import "GDGCondition+Entity.h"
+
+@implementation GDGEntity (SQL)
+
++ (SQLEntityMap *)db
+{
+	@throw [NSException exceptionWithName:@"Abstract Implementation Required"
+	                               reason:@"[GDGEntity+SQL db] throws that 'db' method should be overrided before using any of the categories methods"
+	                             userInfo:nil];
+}
+
++ (void)autoFillProperties:(NSArray <NSString *> *)propertyNames
+{
+	void (^getHandler)(GDGEntity *, NSString *) = ^(GDGEntity *entity, NSString *propertyName) {
+		if (![entity.filledProperties containsObject:propertyName])
+		{
+			[entity fillProperties:@[propertyName]];
+			[entity.filledProperties addObject:propertyName];
+		}
+	};
+
+	void (^setHandler)(GDGEntity *, NSString *) = ^(GDGEntity *entity, NSString *propertyName) {
+		[entity.filledProperties addObject:propertyName];
+		[entity.changedProperties addObject:propertyName];
+	};
+
+	for (NSString *propertyName in propertyNames)
+	{
+		[self addBeforeGetHandler:getHandler forProperty:propertyName];
```

Comment:
Holy shit! I only realize now that anyone can `override` the getter and setter! I dont know if I liked or think this is fucking insane!